unix/capnp_rpc_unix.ml

```diff
@@ -169,14 +169,10 @@ let handle_connection ?tags ~secret_key vat client =
     )
 
 let addr_of_host host =
-  match Unix.gethostbyname host with
-  | exception Not_found ->
-    Capnp_rpc.Debug.failf "Unknown host %S" host
-  | addr ->
-    if Array.length addr.Unix.h_addr_list = 0 then
-      Capnp_rpc.Debug.failf "No addresses found for host name %S" host
-    else
-      addr.Unix.h_addr_list.(0)
+  match Unix.getaddrinfo host "" [Unix.AI_SOCKTYPE Unix.SOCK_STREAM] with
+  | {ai_addr = ADDR_INET(addr, _) ; _} :: _ -> addr
+  | [] -> Capnp_rpc.Debug.failf "No addresses found for host name %S" host
+  | _ -> Capnp_rpc.Debug.failf "Unknown host %S" host
```

Comment:
It's a bit odd to say "unknown host" where there are multiple matches.

Reply:
Indeed this branch was only taken for Unix sockets.
I updated the match to make it explicit.
However, the message is maybe not so relevant anymore. I kept it as it was before, but maybe it deserves to be reworded a bit